client/packages/electron/src/keyboardScanner/keyboardScanner.ts

```diff
@@ -148,17 +149,30 @@ export class KeyboardScanner {
         this.allowEvents = false;
       }
     } catch (e) {
-      console.log(e);
+      console.error(e);
     } finally {
       // Make sure lockBuffer is always unlocked
       this.lockBuffer = false;
     }
   }
 
+  log = (something: string | object) => {
+    try {
+      const js =
+        typeof something === 'object'
+          ? `console.log(${JSON.stringify(something, null, ' ')});`
+          : `console.log('${something.replace("'", "\\'")}');`;
+      this.window.webContents.executeJavaScript(js);
+    } catch (e) {
+      console.error(e);
+    }
+  };
+
   bindInputEvent() {
     // Bind keyboard capture on startup (this.scanning is false by default)
     this.window.webContents.on('before-input-event', (event, input) => {
       if (!this.scanning) return;
+
       if (keysToPassThrough.includes(input.key)) return;
 
       while (this.lockBuffer) {
```

Comment:
The `allowEvents` flag is set to true outside of the keyboard event. The event must be handled on a separate thread?

Reply:
A single JS VM is single threaded and since there is not await point in the loop, this is a busy endless loop.

However, this loop would only become endless if it takes more than 10ms for the keyboard event to reach the handler method (thats how much time we give it to get through before setting `this.allowEvents` to false) BUT we also set `this.lockBuffer` to true in the same micro task meaning we never have the bad condition.

@andreievg Think the while loop is supposed to be just:

```typescript
if (this.allowEvents) return;
```
There is no need to lock the buffer because the buffer access is never interrupted through an await point which means you always have exclusive access to the buffer.

The lock prevents the async `processInputs` method to run concurrently though.